packages/next/server/next-dev-server.ts

```diff
@@ -33,6 +34,19 @@ import HotReloader from './hot-reloader'
 import { findPageFile } from './lib/find-page-file'
 import { getNodeOptionsWithoutInspect } from './lib/utils'
 
+const packageJson = require(pathResolve(__dirname, '../../package.json'))
+
+const currentNodeVersion: string = process.versions.node
+const supportedNodeRange: string = packageJson.engines.node
+
+if (!satisfies(currentNodeVersion, supportedNodeRange)) {
+  const minNodeVersion: SemVer | null = minVersion(supportedNodeRange)
+  throw new Error(
```

Comment:
Let's make this a warning

Reply:
Does a warning really make sense? It's basically guaranteed Next will crash otherwise.

Reply:
It's more so that I don't want to break existing apps that have this issue, I'd rather warn and make it throw in a major. Let's make sure a `TODO:` is added